test/lib/dialogs/question_test.rb

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env rspec
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require_relative "../../test_helper"
+require "autoinstall/dialogs/question"
+require "y2storage/autoinst_issues"
+
+describe Y2Autoinstallation::Dialogs::Question do
+  subject(:dialog) { described_class.new(content, timeout: timeout, buttons_set: buttons_set) }
+
+  let(:timeout) { 0 }
+  let(:content) { "some content" }
+  let(:buttons_set) { :abort }
+
+  before do
+    allow(Yast::UI).to receive(:OpenDialog).and_return(true)
+    allow(Yast::UI).to receive(:CloseDialog).and_return(true)
+  end
+
+  describe "#dialog_content" do
+    it "displays the given content" do
+      expect(dialog.dialog_content.to_s).to include(content)
+    end
+
+    context "when buttons_set is set to :abort" do
+      let(:buttons_set) { :abort }
+
+      it "only shows the 'Abort' button" do
+        expect(dialog).to receive(:PushButton).with(Yast::Term.new(:id, :abort), anything, anything)
```

Comment:
NP: is this actually checking that only abort button is used?

Reply:
More or less: tt is checking that the abort button is being displayed. The problem unit-testing the UI in YaST, is that you cannot stop and check if a given element exists. You need to set expectations like this (a button is created, a widget is updated with 'ChangeWidget`, etc.) and drive the full interaction (mocking `UserInput` or `TimeoutUserInput`).